gpkit/small_classes.py

```diff
@@ -198,20 +199,13 @@ class HashVector(dict):
     >>> exp = gpkit.small_classes.HashVector({x: 2})
     """
 
-    def __init__(self, *args, **kwargs):
-        super(HashVector, self).__init__(*args, **kwargs)
-        self._hashvalue = None
-
     def __hash__(self):
         "Allows HashVectors to be used as dictionary keys."
-        if self._hashvalue is None:
-            self._hashvalue = hash(tuple(self.items()))
+         # pylint:disable=access-member-before-definition, attribute-defined-outside-init
+        if not hasattr(self, "_hashvalue") or self._hashvalue is None:
+            self._hashvalue = reduce(xor, map(hash, self.items()), 0)
```

Comment:
this is much faster, and can be used for incremental updates if we ever implement this in C (not worth it with python function-call overhead)